### PR TITLE
[#176095347] deploy to multiple app services

### PR DESCRIPTION
**Pull request by balanza**
Edit the deploy pipeline to support multiple app services at the same time.

The set of app services to deploy to is stored a map parameter. 
A readonly variable might have been a better fit as parameters can be edited manually before executing the pipeline, however. Azure does not support map in variables. Anyway, **users aren't supposed to edit this value at runtime**.

### Behaviour
* make a release
* build the application once
* deploy and healthcheck on every app service in parallel
* wait for every healthcheck to be executed, then swap every app to production; this means that if only one of the three apps fails on its staging slot, the others won't be promoted to production neither.
Screenshot 2020-12-16 at 15 53 41

### Changes
* App names and resource groups are now hard-coded in the pipeline yaml rather than being provided by pipeline configuration. We decided this makes sense as the infrastructure code as well treat such values as string literals in the codebase. This also means that we might remove unused variables from pipeline configuration when these changes are merged.

### CODE REVIEW

File: .devops/deploy-pipelines.yml [hunk 1]
```diff
@@ -124,49 +142,82 @@ stages:
                 !CODEOWNERS
                 !__*/**/*
             displayName: 'Copy deploy files'
+
+          - script: |
+              ls $(System.DefaultWorkingDirectory)/bundle
+            displayName: 'ls System.DefaultWorkingDirectory/bundle'
+          - script: |
+              ls $(System.DefaultWorkingDirectory)
+            displayName: 'ls System.DefaultWorkingDirectory<'
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:151]:
```suggestion
```
remove only for debug

File: .devops/deploy-pipelines.yml [hunk 2]
```diff
@@ -124,49 +142,82 @@ stages:
...
-              appName: '$(PRODUCTION_APP_NAME)'
-              package: '$(Build.ArtifactStagingDirectory)/'
-              deploymentMethod: 'auto'
-              deployToSlotOrASE: true
-              slotName: 'staging'
-            displayName: Deploy to staging slot
+            - script: |
+                ls $(Pipeline.Workspace)/Bundle
+              displayName: 'ls'
+
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:171]:
```suggestion
```
remove only debug

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

**Comment by pasqualedevita** [comment]
@balanza big work!! 💯

**Comment by gunzip** [comment]
So, is this another pipeline that will live aside the other ones (the actual ones that we use to deploy to a singl app service) ?
I'm asking since the chance to chose just one between the app services to deploy it's a really useful feature.

**Comment by pasqualedevita** [comment]
yes now there are three pipelines, one for each app service.
This pipeline can deploy on all app service by default but if needed we can deploy only to one app service editing this parameter before run the pipeline (for example delete `appAsync` section)
```
app1:
  appname: io-p-fn3-app1
  rg: io-p-rg-functions_app1
app2:
  appname: io-p-fn3-app2
  rg: io-p-rg-internal
```
Screenshot 2020-12-16 at 22 40 27

**Review by gunzip — APPROVED** [review]

**Comment by balanza** [comment]
> So, is this another pipeline that will live aside the other ones (the actual ones that we use to deploy to a singl app service) ?
> I'm asking since the chance to chose just one between the app services to deploy it's a really useful feature.

No, this will replace all the other deploy pipelines and serve as the only entry point for deploying.

I advocate we should always deploy a new version to every production app service.

Even though,  as @pasqualedevita said, it is indeed possible to hack the pipeline to consider only one or two of the app services. Please note that the pipeline also marks a new version of the app, thus we have to take care of this when deploying separately (manually disabling the release stage, probably).

File: .devops/deploy-pipelines.yml (hunk 1 above)
Comment (pasqualedevita) [reply .devops/deploy-pipelines.yml:151]: @balanza can we remove also this?

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

**Review by pasqualedevita — APPROVED** [review]

**Pull request merged**